File: question1/fizzbuzz.rb

def fizz_buzz(n)
    if n % 15 == 0
        'Fizz Buzz'
        elsif n % 3 == 0
        'Fizz'
        elsif n % 5 == 0
        'Buzz'
        else
        n.to_s
    end
end

def countup(m)
    x = 1
    while x < m
        if x == 1
            print fizz_buzz(x)
            else
            print ","
            print fizz_buzz(x)
        end
        x += 1
    end
    puts ""
end

puts "数字を入力してください"
num = gets.to_i
countup(num)

File: question1/fizzbuzz.js
"use strict";

function fizz_buzz(x) {
	if (x % 15 === 0) return "FizzBuzz";
	else if (x % 3 === 0) return "Fizz";
	else if (x % 5 === 0) return "Buzz";
	else return x;
}

function countup(max) {
	var ans = [];
	for (var n = 1; n < max; n++) {
		ans.push(fizz_buzz(n));
	}
	console.log(ans.join(", "));
}

var num = prompt("数字を入力してください");
countup(num);
